emp/custom-registry/emp_image_helper.sh

```diff
@@ -0,0 +1,651 @@
+                VERSION="${img##*:}" # Extract version after the colon
+                CHART="${img%:*}"
+		        name_split "$img"	
+                helm_pull 
+            else
+                $CNT_CMD pull "$img" --platform linux/amd64 >> "$tmpfile" 2>&1 || debug "Failed to pull $img" >> "$tmpfile"
+                info "Pulled $img locally"
+            fi
+        done
+    } &
+
+    # Capture PID of background process
+    local pull_pid=$!
+
+    # Show spinner while pulling
+    spinner "$pull_pid"
+
+    # Check pull results
+    if wait "$pull_pid"; then
+        info "Successfully pulled $num_img images"
+    else
+        warn "Some images failed to pull. Check the log:"
+        cat "$tmpfile" >&2
+    fi
+
+    # Clean up temp file
+    rm "$tmpfile"
+}
+
+# helm_push - Pushes Helm chart to the specified registry
+helm_push() { 
+    local pbs="${CHART_NAME%%/*}"
+    local filename="${CHART_NAME#*/}"
+    local chart_tgz="$filename-$VERSION.tgz"
+    local registry="$1"
+    if [[ -z "$2" ]]; then
+            repo_name="$registry/$2/$pbs"
+        else
+		    repo_name="$registry/$2/$pbs"
+    fi
+    if [[ -z "$chart_tgz" || -z "$registry" ]]; then
+        fail "helm_push: Chart tarball and target registry are required"
+    fi
+    info "Pushing Helm chart: $chart_tgz to $registry"
+    if [[ "$rflag" == 1 ]]; then
+	    helm push "$chart_tgz" oci://"$repo_name" || fail "Failed to push Helm chart $chart_tgz to $registry"
+    else
+	    helm push "$chart_tgz" oci://"$repo_name" || fail "Failed to push Helm chart $chart_tgz to $registry"
+    fi
+    debug "Pushed Helm chart $chart_tgz to $registry"
+}
+
+# push_images_registry - loads container images into given registry
+push_images_registry() {
+    #[ $# -eq 1 ] || fail "push: Please provide registry url"
+    local reg=${1%%/}
+    #echo "$1,$2,$3,$4"
+    local num_img
+    num_imgs=$(num_images)
+    num_img=$(trim_space $num_imgs)
+    info "Pushing $num_img images into $reg container registry/repository..."
+    
+    # Temporary file for output
+    local tmpfile
+    tmpfile=$(mktemp)
+    
+    # Async push with spinner
+    {
+        # Transformation function for image names
+        local trans
+        if echo "$reg" | grep -q /; then
+            trans() {
+                echo "$reg/$(basename "$1")"
+            }
+        else
+            trans() {
+                local i2="${1##docker.io/}" #remove docker.io
+                local i3="${i2##registry.k8s.io/}" #remove registry.k8s.io from i2
+                echo "$reg/${i3##quay.io/}" #remove quay.io from i3
+            }
+        fi
+        
+        # Push images
+        for img in $IMAGES; do
+            local tg
+            name_split "$img"
+            VERSION="${img##*:}"
+            echo "$VERSION"
+            if [[ "$rflag" == 1 ]]; then
+		        aws_create_repo "$3" "$4" "$1" "$2"
+	        else
+		        debug "here in jfrog"
+		        debug "tagging:$tg"
```

Comment:
Remove this line as this this gives error $tg before assign or move tag code above this condition.